### PR TITLE
Changing clusterDomain without downtime

### PR DESCRIPTION
**Pull request by borg-z**
## Description

Updating documentation  and  kubeadm config for seamless `clusterDomain` swapping

## Why do we need it, and what problem does it solve?

The current instruction to change the `clusterDomain` may cause short-term cluster issues. The updated instruction allows you to do this seamlessly without downtime, as described here:

https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#serviceaccount-token-volume-projection

> You can specify the --service-account-issuer argument multiple times, this can be useful to enable a non-disruptive change of the issuer. When this flag is specified multiple times, the first is used to generate tokens and all are used to determine which issuers are accepted. You must be running Kubernetes v1.22 or later to be able to specify --service-account-issuer multiple times.

## What is the expected result?

You can specify an additional service-account-issuer by configuring the control-plane-manager module (`apiserver.serviceAccount.additionalAPIIssuers`).

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi,control-plane-manager,kube-dns
type: fix 
summary: Seamless change of clusterDomain
impact: 
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/openstack'...

File: modules/042-kube-dns/docs/FAQ.md [hunk 1]
```diff
@@ -7,15 +7,43 @@ search: DNS, domain, clusterdomain
...
+**Important!** If your Kubernetes version is 1.20 and higher, your controllers in the cluster use [advanced ServiceAccount tokens](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection) to work with apiserver. Those tokens have extra fields `iss:` and `aud:` that contain `clusterDomain` (e.g. `"iss": "https://kubernetes.default.svc.cluster.local"`). After changing the clusterDomain, the apiserver will start issuing tokens with the new service-account-issuer, but thanks to the configuration of additionalAPIAudiences and additionalAPIIssuers, the apiserver will continue to accept the old tokens.
+After 48 minutes (80% of 3607 seconds), Kubernetes will begin to refresh the issued tokens, and the new service-account-issuer will be used for the updated tokens. After 90 minutes (3607 seconds plus a short buffer) following the kube-apiserver restart, you can remove the serviceAccount configuration from the control-plane-manager configuration.
```
Comment (mvasl) [inline modules/042-kube-dns/docs/FAQ.md:47]: Can we optionally automate this removal?

**Review by mvasl — APPROVED** [review]

File: modules/042-kube-dns/docs/FAQ.md (hunk 1 above)
Comment (borg-z) [reply modules/042-kube-dns/docs/FAQ.md:47]: Yes, but there are two unclear points.
 1. We need to make sure that all tokens have rolled over, not sure if it's right to rely on time here.
 2. We need to store the state before and after the rollover, and we need to account for situations where there might be some changes in the middle of the process.

**Review by name212 — APPROVED** [review]

File: modules/040-control-plane-manager/openapi/config-values.yaml [hunk 2]
```diff
@@ -44,6 +44,12 @@ properties:
               A list of API audiences to add when provisioning ServiceAccount tokens.
             items:
               type: string
+          additionalAPIIssuers:
+            type: array
+            description: |
+              A list of API issuers to add when provisioning ServiceAccount tokens.
```
Comment (nabokihms) [inline modules/040-control-plane-manager/openapi/config-values.yaml:50]: The description has to be detailed. Let's add at least what is in the Kubernetes docs.

> A list of API issuers used to determine which issuers are accepted. This option is beneficial when migrating from one issuer to another.

Comment (borg-z) [reply modules/040-control-plane-manager/openapi/config-values.yaml]: https://github.com/deckhouse/deckhouse/pull/9739/commits/52761c39746a769fd9291f9ccaf26ade54818c2b

**Review by nabokihms — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Lada7878. I've started the workflow for label 'deploy/web/test'...

**Pull request merged**